samples/adk-sql-agent/sql_agent/tools.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+from typing import Any, NotRequired, TypedDict
+from opentelemetry import trace
+from google.adk.tools import ToolContext
+import sqlite3
+import tempfile
+
+SESSION_DB_KEY = "session_sqlite_db"
+
+tracer = trace.get_tracer(__name__)
+logger = logging.getLogger(__name__)
+
+
+class SqlRunResult(TypedDict):
+    error: NotRequired[str]
+    """If the result represents an error. It will be null or undefined if no error"""
+
+    rows: NotRequired[list[tuple[str, ...]]]
+    """The rows returned by the SQL query"""
+
+@tracer.start_as_current_span("create_database")
+def create_database_tool(tool_context: ToolContext):
+    """Creates a temporary file in the /tmp directory to hold an ephemeral
+    sqlite3 database if a database is not found for the current session.
+    """
+    if not SESSION_DB_KEY in tool_context.state:
+        _, path = tempfile.mkstemp(suffix=".db")
+        # No scope prefix in the state data indicates that it will be persisted for
+        # current session.
+        # See https://deepwiki.com/google/adk-python/3.4-state-management.
```

Comment:
Not familiar with DeepWiki, is the URL stable or would it be better to link to https://google.github.io/adk-docs/sessions/state/?

Reply:
I think it's stable - its copied from adk-python's GitHub repo.
However on second thought your suggestion does seem to be the official docs.